draft-ietf-webtrans-http3.md

```diff
@@ -276,27 +276,15 @@ SHALL last until the end of the stream.
 ~~~~~~~~~~
 {: #fig-bidi-client title="WEBTRANSPORT_STREAM frame format"}
 
-## Server-Initiated Bidirectional Streams
+HTTP/3 does not by itself define any semantics for server-initiated
+bidirectional streams.  If WebTransport setting is negotiated by both
+endpoints, the syntax of the server-initiated bidirectional streams SHALL be
+the same as the syntax of client-initated bidirectional streams, that is, a
+sequence of HTTP/3 frames.  The only allowed frame is WEBTRANSPORT_STREAM;
```

Comment:
I would suggest instead allowing all HTTP/3 frame types, and saying that clients MUST silently ignore all frame types except for WEBTRANSPORT_STREAM. That'll enable future extensibility without requiring a SETTING. This will also make it easier to split this into its own draft.

Reply:
But this frame is a TV frame (not TLV), so it's defined to take up the entire stream?  

To get more extensibility, we could have server-initiated bidi stream carry a 'Stream Type', like uni streams, followed by a series of HTTP/3 frames.  WebTransport would then define its TV frame that converts the stream to unframed, and other types of server-initiated bidi streams could use TLV or TV frames.

Reply:
The frame being TV means that it needs to be the *last* frame on a stream. It doesn't need to be the *only* frame on a stream. All I'm saying is that server-initiated bidi streams can carry any and all h3 frames, and when they see a WEBTRANSPORT_STREAM frame then the rest of the stream is taken over. Just like client-initiated bidi streams.

Reply:
The frame is currently defined to be both first (sets the type to WT) and last (it's TV).  

I'm suggesting you might want to split those two functions, and all streams (except client-initiated bidi) start with a stream type, followed by frames.

Reply:
I don't think we want to allow _all_ frame types. In particular, allowing HEADERS to be sent and but dropped at the receiver seems like a good way to break QPACK (see https://datatracker.ietf.org/doc/html/draft-ietf-quic-qpack-21#section-2.2.2.1). Opening things up for other frames types, especially new extension frames, seems ok though.

Reply:
@afrind what does "sets the type" mean? bidi streams don't have a type. It makes sense to allow extension frames before jumping into WT.

@LPardue that's a good point, maybe not all frames, but definitely extensions

Reply:
@DavidSchinazi : This draft defines a frame with two fields - one is the frame type, which serves multiple functions, and the session ID.  As written, sending this frame identifies the stream as a WT stream (semantics governed by this draft), associates this stream with a WT session, and converts the stream to unframed.

For general server bidi-stream extension semantics, I prefer we send a single 'type' field (not a frame type) to identify the semantics of the stream, followed by frames.  Then this draft would be slightly different:

```
[ Stream Type=WT Server-Init Bidi Stream ] [ Frame Type=WT_STREAM | Session ID ]
```

@LPardue : I don't think we should disallow any QPACK bearing frames in extensions, any more than H2 forbid HPACK bearing frames.  You just have to ensure that the peer has expressed support for the extension before using the dynamic table.  The static table is always OK.

Reply:
oh I'm not trying to prevent new things that can carry QPACK, I just think you'll want to couple that with a negotiation that makes it explicit the other end would respect the rules. Discarding HEADERS seems like an odd choice of behaviour to enable.

Reply:
It sounds like this question needs further discussion.  I am leaving the frame-based version here to get us to a draft we can interop on, but we can continue this discussion on issue #23 (and at the upcoming interim).